### PR TITLE
chore: Add remove backend wallet option

### PR DESCRIPTION
**Pull request by arcoraven**
## PR-Codex overview
The focus of this PR is to add functionality related to backend wallets, including removing wallets and handling transactions.

### Detailed summary
- Added `removeBackendWallet` function to delete wallet details
- Updated backend wallet routes to include new functionalities like `sendTransactionBatch`, `signMessage`, `simulateTransaction`
- Added configuration updates related to backend wallet balances and CORS
- Included new contract events handling functions

> The following files were skipped due to too many changes: `src/server/routes/index.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/server/routes/index.ts [hunk 1]
```diff
@@ -1,58 +1,35 @@
...
-import { setDefaultRoyaltyInfo } from "./contract/royalties/write/setDefaultRoyaltyInfo";
-import { setTokenRoyaltyInfo } from "./contract/royalties/write/setTokenRoyaltyInfo";
-
-// Contract metadata
-import { getABI } from "./contract/metadata/abi";
-import { extractEvents } from "./contract/metadata/events";
-import { extractFunctions } from "./contract/metadata/functions";
-
-// Wallet
+import { createAccessToken } from "./auth/access-tokens/create";
```
Comment (arcoraven) [inline src/server/routes/index.ts:2]: I'm unsorting the imports here. I believe all devs should be using auto-imports from our IDE which will just add/remove it as needed. It's not smart enough to put imports in the right section, nor do we really care.

Comment (d4mr) [reply src/server/routes/index.ts:2]: we should be auto generating barrel files using something like [barrelsby](https://www.npmjs.com/package/barrelsby)

File: src/server/routes/backend-wallet/remove.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import { Static, Type } from "@sinclair/typebox";
+import { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import { Address } from "thirdweb";
+import { deleteWalletDetails } from "../../../db/wallets/deleteWalletDetails";
+import { createCustomError } from "../../middleware/error";
+import { standardResponseSchema } from "../../schemas/sharedApiSchemas";
+
+const requestSchema = Type.Object({
+  walletAddress: Type.Optional(Type.String()),
```
Comment (d4mr) [inline src/server/routes/backend-wallet/remove.ts:10]: why do we make it optional here, but validate its existence later?

File: src/server/routes/backend-wallet/remove.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      const { walletAddress: _walletAddress } = req.params;
+      if (!_walletAddress) {
+        throw createCustomError(
+          'Missing "walletAddress".',
+          StatusCodes.BAD_REQUEST,
+          "BAD_REQUEST",
+        );
+      }
+
+      const walletAddress = _walletAddress.toLowerCase() as Address;
```
Comment (farhanW3) [inline src/server/routes/backend-wallet/remove.ts:54]: the `deleteWalletAddress()` already does a `toLowerCase()`, do we need it here too?

File: src/server/routes/backend-wallet/remove.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    Reply: Static<typeof responseSchema>;
+  }>({
+    method: "DELETE",
+    url: "/backend-wallet/:walletAddress",
+    schema: {
+      summary: "Remove backend wallet",
+      description:
+        "Remove an existing backend wallet. NOTE: This is an irreversible action for local wallets. Ensure any funds are transferred out before removing a local wallet.",
+      tags: ["Backend Wallet"],
+      operationId: "remove",
```
Comment (farhanW3) [inline src/server/routes/backend-wallet/remove.ts:37]: nit: can we update `operationId` to `removeBackendWallet`.

File: src/server/routes/backend-wallet/remove.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import { Static, Type } from "@sinclair/typebox";
+import { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import { Address } from "thirdweb";
+import { deleteWalletDetails } from "../../../db/wallets/deleteWalletDetails";
+import { createCustomError } from "../../middleware/error";
+import { standardResponseSchema } from "../../schemas/sharedApiSchemas";
+
+const requestSchema = Type.Object({
```
Comment (farhanW3) [inline src/server/routes/backend-wallet/remove.ts:9]: nit: update it to `requestParamSchema` to be precise.

File: src/server/routes/backend-wallet/remove.ts (hunk 2 above)
Comment (arcoraven) [reply src/server/routes/backend-wallet/remove.ts:10]: Hmm you're right I copied it from a different route. Should just be required.

**Review by farhanW3 — APPROVED** [review]
lgtm

**Pull request merged**